File: DesignPatternsExercises/java/StateDesignPattern/Solution/StateDesignPattern.java
public class StateDesignPattern {
    public static void main(String[] args) {
        AtmMachine atmMachine = new AtmMachine();

        System.out.println("ATM Machine current state: " + atmMachine.getAtmMachineState());
        atmMachine.withdraw();
        atmMachine.ejectCard();

        System.out.println("\nATM Machine current state: " + atmMachine.getAtmMachineState());
        atmMachine.insertCard();

        System.out.println("\nATM Machine current state: " + atmMachine.getAtmMachineState());
        atmMachine.withdraw();
        atmMachine.ejectCard();
    }
}

File: DesignPatternsExercises/ruby/state_design_pattern/exercise/state_design_pattern.rb
require_relative 'atm_machine'

atm_machine = AtmMachine.new

puts "ATM Machine current state: #{atm_machine.atm_machine_state}"
atm_machine.withdraw?
atm_machine.eject_card?

puts "\nATM Machine current state: #{atm_machine.atm_machine_state}"
atm_machine.insert_card?

puts "\nATM Machine current state: #{atm_machine.atm_machine_state}"
atm_machine.withdraw?
atm_machine.eject_card?
